src/app/auth/auth.js

```diff
@@ -0,0 +1,152 @@
+angular.module('prx.auth', ['prx.ui.nav'])
+.config(function (PRXDrawerProvider) {
+  PRXDrawerProvider.register({
+    name: 'Sign In',
+    weight: PRXDrawerProvider.BOTTOM,
+    href: 'http://www.prx.org/sessions/new',
+    nav: true,
+    template: '<prx-auth-badge></prx-auth-badge>',
+    dropdownTemplate: '<prx-auth-window></prx-auth-window>'
+  });
+}).service('PrxAuth', function (ngHal, $q, $window) {
+  var PrxAuth = this,
+      deferred = $q.defer(),
+      currentUser = {loggedIn: false};
+  this.$$loginPromise = deferred.promise;
+
+  this.$checkLoggedIn = function () {
+    var iframe = angular.element('<iframe>'),
+        $body = angular.element($window.document.body),
+        uri = [FEAT.ID_SERVER + '/authorize?client_id=' + FEAT.ID_CLIENT_KEY],
+        nonce = [];
+    for (var i=0; i<8; i++) {
+      nonce.push(randomInt(0, 15).toString(16));
+    }
+    nonce = nonce.join('');
+    uri.push("&nonce=", nonce, "&response_type=token&prompt=none");
+    iframe.css('display', 'none');
+    iframe.attr('src', uri.join(''));
+
+    iframe.on('load', function onLoad() {
+      var query = false, accessToken;
+      try {
+        query = iframe[0].contentDocument.location.hash.replace(/^#/,'');
+      } catch (e) {}
+      if (query && (accessToken = parseQuery(query).access_token)) {
+        PrxAuth.$processToken(accessToken).then(function (user) {
+          deferred.resolve(user);
+        });
+      } else {
+        deferred.reject('not logged in');
+      }
+      iframe.off('load', onLoad);
+      iframe.remove();
+      iframe = undefined;
+    });
+
+    $body.append(iframe);
+  };
+
+  this.$processToken = function (token) {
+    this.token = token;
+    this.loggedIn = true;
+    return ngHal.follow('prx:authorization', {}, headers(token)).then(function (result) {
```

Comment:
first, we make a request to the CMS API to get information about the currently active authorization